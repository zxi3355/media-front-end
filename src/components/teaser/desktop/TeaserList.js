import React from 'react';
import Advertisement from '../../Advertisement';
import TeaserItem from './TeaserItem';
import '../TeaserList.common.scss';

const TeaserList = ({ teasers }) => {
  return (
    <div className='TeaserList'>
      <div className='TeaserList--teasers'>
        {teasers.map((teaser) => (
          <TeaserItem key={teaser.id} teaser={teaser} />
        ))}
      </div>
      <div className='TeaserList--ads'>
        <Advertisement />
      </div>
    </div>
  );
};

export default TeaserList;
