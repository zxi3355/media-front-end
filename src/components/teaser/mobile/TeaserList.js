import React from 'react';
import Advertisement from '../../Advertisement';
import TeaserItem from './TeaserItem';
import './TeaserList.mobile.scss';

const TeaserList = ({ teasers }) => {
  return (
    <div className='TeaserListMobile'>
      <Advertisement />
      <div>
        {teasers.map((teaser) => (
          <TeaserItem key={teaser.id} teaser={teaser} />
        ))}
      </div>
    </div>
  );
};

export default TeaserList;
