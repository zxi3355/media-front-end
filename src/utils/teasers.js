export const TEASERS = {
  totalCount: 24,
  data: [
    {
      id: 'NOW-54531',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/the-queen-instagram-54531',
      source: "Australian Women's Weekly",
      contentTitle: 'Queen Elizabeth has Instagrammed for the first time',
      contentSummary: 'Yass Queen!',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/08/54531/Hero1.jpg',
      pageDateCreated: '2019-03-07T20:27:48.00Z',
    },
    {
      id: 'NOW-54515',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/kate-middleton-family-holiday-blackpool-54515',
      source: "Australian Women's Weekly",
      contentTitle:
        'Duchess Catherine just hinted where the next Cambridge family holiday will be',
      contentSummary:
        'On a rainy day in Blackpool, the Duke and Duchess of Cambridge brought some brightness.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/07/54515/HERO.jpg',
      pageDateCreated: '2019-03-06T20:37:57.00Z',
    },
    {
      id: 'NOW-54514',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-prince-harry-stage-we-day-54514',
      source: "Australian Women's Weekly",
      contentTitle:
        'Royal PDA! Duchess Meghan makes surprise appearance with Prince Harry in an adorable on-stage reunion',
      contentSummary:
        'It would be safe to say these two are *very* excited to become parents.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/07/54514/HERO.jpg',
      pageDateCreated: '2019-03-06T20:02:42.00Z',
    },
    {
      id: 'NOW-51842',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-maternity-fashion-51842',
      source: "Australian Women's Weekly",
      contentTitle: "Duchess Meghan Markle's best maternity fashion moments",
      contentSummary: 'That is one chic baby bump.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/06/51842/HERO-March-6.jpg',
      pageDateCreated: '2019-03-06T00:02:00.00Z',
    },
    {
      id: 'NOW-54479',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-kate-middleton-investiture-54479',
      source: "Australian Women's Weekly",
      contentTitle:
        'Duchess Meghan and Duchess Catherine just stepped out together in a dazzling display',
      contentSummary: 'The pair united for a cause very close to home.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/06/54479/HERO-1.jpg',
      pageDateCreated: '2019-03-05T19:56:38.00Z',
    },
    {
      id: 'NOW-54464',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-maternity-leave-54464',
      source: "Australian Women's Weekly",
      contentTitle:
        "Is this Duchess Meghan's final appearance before she goes on maternity leave?",
      contentSummary:
        'The hard-working Duchess has just announced another upcoming appearance - but will it be her last before she gears up to welcome her first child?',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/05/54464/MEGHAN-MARKLE-MATERNITY-LEAVE.jpg',
      pageDateCreated: '2019-03-05T01:30:00.00Z',
    },
    {
      id: 'NOW-54458',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/new-royal-family-social-media-rules-54458',
      source: "Australian Women's Weekly",
      contentTitle:
        'How the royal family is taking action against cyberbullies on social media',
      contentSummary:
        'The palace is taking action after members of the royal family, particularly Duchess Catherine and Duchess Meghan, were bullied online.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/05/54458/HERO.jpg',
      pageDateCreated: '2019-03-04T20:48:43.00Z',
    },
    {
      id: 'NOW-54433',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/prince-harry-meghan-markle-royal-baby-gender-fluid-54433',
      source: "Australian Women's Weekly",
      contentTitle:
        "Kensington Palace have made a rare statement about Harry and Meghan's baby's gender",
      contentSummary:
        'A magazine article has published something "totally false" about the Sussexes\' parenting plans. And now, the royal household is speaking out.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/04/54433/HERO.jpg',
      pageDateCreated: '2019-03-03T21:57:14.00Z',
    },
    {
      id: 'NOW-54430',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-mother-doria-visit-la-54430',
      source: 'Now to Love',
      contentTitle:
        'Meghan Markle has secretly been flying to visit to her mother in LA for a very important reason',
      contentSummary: 'Doria to the rescue.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/04/54430/HERO.jpg',
      pageDateCreated: '2019-03-03T20:20:28.00Z',
    },
    {
      id: 'NOW-54393',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-divorce-plan-54393',
      source: "Woman's Day",
      contentTitle:
        "Meghan Markle's divorce back-up plan revealed: A $25 million New York City escape",
      contentSummary:
        "''Sheâ€™s got an entire back-up plan arranged for when and if her and Harry call it quits.''",
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/01/54393/HERO.jpg',
      pageDateCreated: '2019-03-03T20:00:00.00Z',
    },
    {
      id: 'NOW-54404',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-baby-shower-goodie-bag-54404',
      source: "Woman's Day",
      contentTitle: "Inside Meghan Markle's expensive baby shower goodie bags",
      contentSummary: "Only the best for Meghan's A-list pals!",
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/01/54404/Hero1.jpg',
      pageDateCreated: '2019-02-28T23:20:27.00Z',
    },
    {
      id: 'NOW-54388',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/kate-middleton-more-children-54388',
      source: "Australian Women's Weekly",
      contentTitle:
        'Will there be a fourth Cambridge baby? Duchess Catherine gives her answer',
      contentSummary:
        'Catherine is clucky! And she revealed what Prince William thinks of bringing another royal baby into the world.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/01/54388/Hero2.jpg',
      pageDateCreated: '2019-02-28T20:45:43.00Z',
    },
    {
      id: 'NOW-54360',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/kate-middleton-fashion-belfast-54360',
      source: "Australian Women's Weekly",
      contentTitle:
        'Duchess Catherine shows her sporty and chic sides on official trip to Belfast with Prince William',
      contentSummary: 'Kate the Great is back at it again!',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/28/54360/HERO-(1).jpg',
      pageDateCreated: '2019-02-27T20:30:20.00Z',
    },
    {
      id: 'NOW-54335',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/princess-diana-musical-reaction-54335',
      source: 'Now to Love',
      contentTitle:
        'A new musical about Princess Diana has been torn to shreds by royal commentators: "It\'s just bad taste"',
      contentSummary:
        'And one particular scene has *really* riled up royal fans.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/27/54335/HERO-2.jpg',
      pageDateCreated: '2019-02-27T00:11:00.00Z',
    },
    {
      id: 'NOW-50702',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/royal-curtsies-50702',
      source: 'Now to Love',
      contentTitle:
        'The best royal curtsies of all time because you need to show The Queen how low you can go',
      contentSummary:
        "It's the royal version of the bend and snap - and it's drenched with history and protocol.",
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/27/50702/ROYAL-CURTSIES.jpg',
      pageDateCreated: '2019-02-27T00:00:00.00Z',
      contentImageAltText: 'royal curtsies',
    },
    {
      id: 'NOW-54331',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/queen-meghan-markle-opinion-54331',
      source: "Australian Women's Weekly",
      contentTitle:
        "The Queen just broke her silence on Duchess Meghan's latest public appearances",
      contentSummary: "She's charmed the world, but what does the Queen think?",
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/27/54331/HERO.jpg',
      pageDateCreated: '2019-02-26T20:07:45.00Z',
    },
    {
      id: 'NOW-54317',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/royal-scandals-54317',
      source: "Woman's Day",
      contentTitle: 'The 50 biggest royal scandals of all time',
      contentSummary:
        'The House of Windsor is always making headlines - sometimes for the wrong reasons.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/26/54317/HERO.jpg',
      pageDateCreated: '2019-02-26T05:00:00.00Z',
    },
    {
      id: 'NOW-54318',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/kate-middleton-meghan-markle-joint-appearance-54318',
      source: "Australian Women's Weekly",
      contentTitle:
        'Duchesses Meghan and Kate are about to make a rare joint appearance together for a surprising reason',
      contentSummary:
        'The royal sisters-in-law are coming together for a very special reason.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/26/54318/HERO.jpg',
      pageDateCreated: '2019-02-26T00:12:00.00Z',
    },
    {
      id: 'NOW-54319',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/kate-middleton-prince-george-princess-charlotte-pool-54319',
      source: "Australian Women's Weekly",
      contentTitle:
        'Duchess Catherine just treated Prince George and Princess Charlotte to the BEST day out',
      contentSummary: 'The Cambridge clan make a splash!',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/26/54319/KATE-MIDDLETON-PRINCESS-CHARLOTTE-PRINCE-GEORGE.jpg',
      pageDateCreated: '2019-02-25T23:41:00.00Z',
    },
    {
      id: 'NOW-54315',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-morocco-outfit-fashion-54315',
      source: "Australian Women's Weekly",
      contentTitle:
        "Meghan's Moroccan fashion show: The Duchess wears THREE different maternity looks in one day",
      contentSummary:
        'It might be fashion week in Paris, but Meghan was bringing all the style to Morocco.',
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/26/54315/HERO.jpg',
      pageDateCreated: '2019-02-25T20:32:44.00Z',
    },
    {
      id: 'NOW-54297',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-prince-harry-morocco-54297',
      source: "Australian Women's Weekly",
      contentTitle:
        'Heavily pregnant Duchess Meghan dazzles as she tours Morocco with Prince Harry',
      contentSummary:
        "It's their one last hurrah before they become parents, and they're not holding back!",
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/25/54297/HERO.jpg',
      pageDateCreated: '2019-02-24T22:10:00.00Z',
    },
    {
      id: 'NOW-54285',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/queen-confides-kate-middleton-54285',
      source: "Woman's Day",
      contentTitle:
        "Queen confides in Duchess Catherine... as 'Hurricane Meghan' rages",
      contentSummary:
        "The two duchesses' true colours come shining through as the Queen turns to Kate with her concerns.",
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/22/54285/KATE-MIDDLETON-MEGHAN-MARKLE.jpg',
      pageDateCreated: '2019-02-24T20:00:00.00Z',
    },
    {
      id: 'NOW-54284',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/meghan-markle-mummy-necklace-54284',
      source: "Australian Women's Weekly",
      contentTitle:
        "Duchess Meghan wore an adorable necklace - and we've found the budget version!",
      contentSummary:
        "The necklace is a sweet nod to her new role as a mother and it's super cute!",
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/22/54284/Hero.jpg',
      pageDateCreated: '2019-02-22T01:18:21.00Z',
    },
    {
      id: 'NOW-54278',
      url: 'https://www.nowtolove.com.au/royals/british-royal-family/kate-middleton-prince-william-snow-holiday-54278',
      source: 'Now to Love',
      contentTitle:
        "Kate Middleton skips out on Meghan's baby shower for family snow holiday",
      contentSummary:
        "Now we know why Kate was absent from Meghan's big 'do in New York City.",
      contentImageUrl:
        'http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/02/22/54278/Hero2.jpg',
      pageDateCreated: '2019-02-21T22:23:24.00Z',
    },
  ],
};
