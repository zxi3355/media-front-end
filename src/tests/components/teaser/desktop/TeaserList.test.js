import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import teasers from '../../../fixtures/teasers';
import TeaserList from '../../../../components/teaser/desktop/TeaserList';

test('Should render TeaserList page correctly', () => {
  const wrapper = shallow(<TeaserList teasers={teasers} />);
  expect(wrapper.find('.TeaserList').length).toEqual(1);
  expect(toJson(wrapper)).toMatchSnapshot();
});
