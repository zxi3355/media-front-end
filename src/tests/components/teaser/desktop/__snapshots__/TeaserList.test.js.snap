// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TeaserList page correctly 1`] = `
<div
  className="TeaserList"
>
  <div
    className="TeaserList--teasers"
  >
    <TeaserItem
      key="NOW-54531"
      teaser={
        Object {
          "contentImageUrl": "http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/08/54531/Hero1.jpg",
          "contentSummary": "Yass Queen!",
          "contentTitle": "Queen Elizabeth has Instagrammed for the first time",
          "id": "NOW-54531",
          "pageDateCreated": "2019-03-07T20:27:48.00Z",
          "source": "Australian Women's Weekly",
          "url": "https://www.nowtolove.com.au/royals/british-royal-family/the-queen-instagram-54531",
        }
      }
    />
    <TeaserItem
      key="NOW-54515"
      teaser={
        Object {
          "contentImageUrl": "http://d3lp4xedbqa8a5.cloudfront.net/s3/digital-cougar-assets/Now/2019/03/07/54515/HERO.jpg",
          "contentSummary": "On a rainy day in Blackpool, the Duke and Duchess of Cambridge brought some brightness.",
          "contentTitle": "Duchess Catherine just hinted where the next Cambridge family holiday will be",
          "id": "NOW-54515",
          "pageDateCreated": "2019-03-06T20:37:57.00Z",
          "source": "Australian Women's Weekly",
          "url": "https://www.nowtolove.com.au/royals/british-royal-family/kate-middleton-family-holiday-blackpool-54515",
        }
      }
    />
  </div>
  <div
    className="TeaserList--ads"
  >
    <Advertisement />
  </div>
</div>
`;
